src/HTTP.jl

```diff
@@ -626,12 +636,54 @@ function stack(;redirect=true,
     (verbose >= 3 ||
      DEBUG_LEVEL[] >= 3   ? DebugLayer                : NoLayer){
     (readtimeout > 0      ? TimeoutLayer              : NoLayer){
-                            StreamLayer
+                            StreamLayer{Union{}}
     }}}}}}}}}}}}
 end
 
-include("download.jl")
 
+"""
+    next(::Type{S}) where {T, S<:Layer{T}}
+
+Return the next layer in the stack
+"""
+function next(::Type{S}) where {T, S<:Layer{T}}
+    return T
+end
+
+"""
+    head(::Type{T}) where T <: Layer
+
+Get the name of the type
+"""
+function head(::Type{T}) where T <: Layer
```

Comment:
Why is this called `head` ?
why not say `wrapper_type`

Reply:
`top_layer` maybe, given that this is specialized on Layers. It doesn't need to be generic to all types.